### PR TITLE
Account runtypes validation

### PR DESCRIPTION
**Pull request by qbob1**
Implemented run time validation during api requests for the account component using the lib runtypes.

### CODE REVIEW

File: angular/src/app/modules/account/account/account.component.ts [hunk 1]
```diff
@@ -24,7 +25,8 @@ export class AccountComponent implements OnInit {
   constructor(private readonly accountService: AccountService) {}
 
   ngOnInit(): void {
-    this.account$ = this.accountService.get('100');
+    this.account$ = this.accountService.get('100').pipe(filter((account) => TypeGuard(account)));
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.ts:28]: this kind of validation should run in the account service, not here in the component

File: angular/src/app/modules/account/typeguard.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  phone: String,
+});
+const Account = Record({
+  address: Address,
+  payments: Array(Payment),
+  profiles: Array(Profile),
+});
+const TypeGuard = (el: Account | Acct | {}) => Account.guard(el);
+
+export default TypeGuard;
```
Comment (escalonn) [inline angular/src/app/modules/account/typeguard.ts:36]: call this AccountTypeGuard (& change filename too, to `account-type-guard`), not just TypeGuard
normally i'd say most of this should be camelCase, but we can follow the runtypes docs and keep TitleCase for variables that represent types

Comment (qbob1) [reply angular/src/app/modules/account/typeguard.ts]: commit:d0560ba

File: angular/src/app/modules/account/account-type-guard.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Account as Acct } from '../../data/account.model';
```
Comment (escalonn) [inline angular/src/app/modules/account/account-type-guard.ts:1]: what's the purpose of this import?

**Comment by escalonn** [comment]
do it this way https://github.com/pelotom/runtypes#static-type-inference

**Comment by escalonn** [comment]
well, actually, i didnt pay enough attention to that... if you can do `interface` instead of `type` there, then _maybe_ that'd work... ideally there'd just be a way to point runtypes (or some other dependency) to our existing typescript interfaces and have them checked purely based on that without having to repeat ourselves. it's bad enough with those jsons, it'd be ideal if we didnt have to maintain yet another thing that needs to be consistent with the types

**Comment by fredbelotte** [comment]
where are we with this PR?

**Comment by escalonn** [comment]
interesting attempt but implementation didnt get far enough that we won't just close it.
if we wanted to do this again, i'd look at a different library than "runtypes"... the goal would be to follow DRY and not have a separate definition of the types for runtime.

**Pull request closed without merging**